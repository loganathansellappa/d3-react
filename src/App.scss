.app-component {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  .header {
    background-color: #007bff;
    color: white;
    padding: 10px;
    width: 100%;
    box-sizing: border-box;
    align-self: flex-start;
  }

  .content {
    margin: 10px;
    flex: 1;
    display: flex;
    flex-direction: column;

    table {
      width: 50%;
      border-collapse: collapse;
      background-color: #fff;
      box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1);

      th,
      td {
        padding: 12px 15px;
        text-align: left;
      }

      th {
        background-color: #007bff;
        color: white;
      }

      tr:nth-child(even) {
        background-color: #f2f2f2;
      }

      tr:hover {
        background-color: #ddd;
      }
      .bold-property {
        font-weight: bold; /* Add this to make the first cell bolder */
      }
    }
  }

  .footer {
    background-color: #007bff;
    color: white;
    padding: 10px;
    width: 100%;
    box-sizing: border-box;
    align-self: flex-end;
  }
}
