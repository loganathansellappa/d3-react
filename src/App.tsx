import React from "react";
import {BrowserRouter,  Route, Routes} from "react-router-dom";
import './App.scss';
import {Tabs} from "./components/Tabs";
import {Overview} from "./components/Overview";
import {IncomeStatement} from "./components/IncomeStatement";
import {BalanceSheet} from "./components/BalanceSheet";
import {ContextMenuProvider} from "./components/ContextMenuProvider";
import {QueryClient, QueryClientProvider} from "react-query";
import {Chart} from "./components/Chart";

const queryClient = new QueryClient();

const App: React.FC = () => {
                return (
                <BrowserRouter>
                    <QueryClientProvider client={queryClient}>
                    <ContextMenuProvider>
                        <div className="app-component">
                            <Tabs />
                            <div className="content">
                                <Routes>
                                    <Route path="/overview" index element={<Overview/>} />
                                    <Route path="/balance" element={<BalanceSheet />} />
                                    <Route path="/income" element={<IncomeStatement />} />
                                    <Route path="/chart" element={<Chart />} />
                                </Routes>
                            </div>
                            <footer className="footer">Footer</footer>
                        </div>
                    </ContextMenuProvider>
                        </QueryClientProvider>
                </BrowserRouter>

    );
};

export default App;
