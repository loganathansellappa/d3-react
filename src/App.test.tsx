import App from "./App";
import { render } from "@testing-library/react";

let element: { container: HTMLElement };

test("renders App component", () => {
  element = render(<App />);
  expect(element.container.firstChild).toMatchSnapshot();
  expect(element.container.textContent).toContain("Overview");
});
