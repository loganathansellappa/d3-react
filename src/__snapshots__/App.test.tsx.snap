// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div>
  <div
    class="app-component"
  >
    <div
      class="tabs"
    >
      <a
        class="tab-link "
        href="/overview"
      >
        Overview
      </a>
      <a
        class="tab-link "
        href="/balance"
      >
        Balance Sheet
      </a>
      <a
        class="tab-link "
        href="/income"
      >
        Income Statement
      </a>
      <a
        class="tab-link "
        href="/chart"
      >
        Chart
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="spinner-container"
      >
        <div
          class="loading-spinner"
          style="border: 4px solid #0073e6; border-top: 4px solid transparent; border-radius: 50%; width: 40px; height: 40px; animation: spin 2s linear infinite;"
        />
      </div>
    </div>
    <footer
      class="footer"
    >
      @Demo App 
    </footer>
  </div>
</div>
`;
